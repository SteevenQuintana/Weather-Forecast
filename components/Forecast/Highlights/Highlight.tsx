import Image from 'next/image'

interface HighlightProps {
  title: string
  img: string
  description: string
  unit: string
}

const Highlight = ({ title, img, description, unit }: HighlightProps) => (
  <div className='w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)] hover:cursor-pointer'>
    <h3>{title}</h3>
    <Image
      src={img}
      alt={description}
      height={150}
      width={150}
      className='m-auto'
    />
    <p>
      {description} <span>{unit}</span>
    </p>
  </div>
)

export default Highlight
