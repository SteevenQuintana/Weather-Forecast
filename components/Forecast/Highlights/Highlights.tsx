import humidity from '@/public/assets/humidity.svg'
import wind from '@/public/assets/wind.svg'
import sunrise from '@/public/assets/sunrise.svg'
import sunset from '@/public/assets/sunset.svg'
import clouds from '@/public/assets/clouds.svg'
import pressure from '@/public/assets/pressure.svg'
import Highlight from './Highlight'

import { ForecastProps } from '@/interfaces/cities.interface'
import { getHour } from '@/helper/format'

const Highlights = ({ forecast }: ForecastProps) => {
  if (forecast === null) return
  return (
    <div className='h-full mx-4 md:mx-0 mb-8 text-center flex flex-col bg-[#3054b7ca] p-4 rounded-2xl dark:bg-[#3a4957cc]'>
      <h2 className='text-2md font-bold mb-4'>Today's Highlights</h2>
      <div className='grid grid-cols-[repeat(auto-fill,minmax(160px,1fr))] gap-3 h-full'>
        <Highlight
          title='Humidity'
          img={humidity}
          description={forecast.main.humidity}
          unit='%'
        />
        <Highlight
          title='Wind Speed'
          img={wind}
          description={forecast.wind.speed}
          unit='m/s'
        />
        <Highlight
          title='Clouds'
          img={clouds}
          description={forecast.clouds}
          unit='%'
        />
        <Highlight
          title='Sunrise'
          img={sunrise}
          description={getHour(forecast.sunrise)}
          unit='a.m.'
        />
        <Highlight
          title='Sunset'
          img={sunset}
          description={getHour(forecast.sunset)}
          unit='p.m.'
        />
        <Highlight
          title='Pressure'
          img={pressure}
          description={forecast.main.pressure}
          unit='hPa'
        />
      </div>
    </div>
  )
}

export default Highlights
