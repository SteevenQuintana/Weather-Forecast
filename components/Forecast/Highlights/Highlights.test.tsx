import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import { Forecast } from '@/interfaces/cities.interface'
import Highlights from './Highlights'

const mockForecast: Forecast = {
  name: 'City Name',
  country: 'Country Name',
  weather: {
    description: 'Sunny',
    main: 'Clear',
    icon: '01d'
  },
  sunrise: 92291,
  sunset: 91291,
  clouds: 91,
  main: {
    temp: 25,
    temp_max: 30,
    temp_min: 20,
    pressure: 1010,
    humidity: 60,
    feels_like: 26
  },
  wind: {
    speed: 5,
    deg: 90
  }
}

describe('Highlights', () => {
  it('should render the Highlights component', () => {
    render(<Highlights forecast={mockForecast} />)

    expect(screen.getByText("Today's Highlights")).toBeInTheDocument()
    expect(screen.getByText('Humidity')).toBeInTheDocument()
    expect(screen.getByText('Wind Speed')).toBeInTheDocument()
    expect(screen.getByText('a.m.')).toBeInTheDocument()
  })

  it('should snapshots the Highlights component', () => {
    const { container } = render(<Highlights forecast={mockForecast} />)
    expect(container).toMatchSnapshot()
  })
})
