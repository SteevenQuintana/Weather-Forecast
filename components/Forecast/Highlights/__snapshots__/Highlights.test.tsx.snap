// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlights should snapshots the Highlights component 1`] = `
<div>
  <div
    class="h-full mb-8 text-center flex flex-col bg-[#3054b7ca] p-4 rounded-2xl dark:bg-[#3a4957cc]"
  >
    <h2
      class="text-2md font-bold mb-4"
    >
      Today's Highlights
    </h2>
    <div
      class="grid grid-cols-[repeat(auto-fill,minmax(160px,1fr))] gap-3 h-full"
    >
      <div
        class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
      >
        <h3>
          Humidity
        </h3>
        <img
          alt="60"
          class="m-auto"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <p>
          60
           
          <span>
            %
          </span>
        </p>
      </div>
      <div
        class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
      >
        <h3>
          Wind Speed
        </h3>
        <img
          alt="5"
          class="m-auto"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <p>
          5
           
          <span>
            m/s
          </span>
        </p>
      </div>
      <div
        class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
      >
        <h3>
          Clouds
        </h3>
        <img
          alt="91"
          class="m-auto"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <p>
          91
           
          <span>
            %
          </span>
        </p>
      </div>
      <div
        class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
      >
        <h3>
          Sunrise
        </h3>
        <img
          alt="20:38"
          class="m-auto"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <p>
          20:38
           
          <span>
            a.m.
          </span>
        </p>
      </div>
      <div
        class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
      >
        <h3>
          Sunset
        </h3>
        <img
          alt="20:21"
          class="m-auto"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <p>
          20:21
           
          <span>
            p.m.
          </span>
        </p>
      </div>
      <div
        class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
      >
        <h3>
          Pressure
        </h3>
        <img
          alt="1010"
          class="m-auto"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <p>
          1010
           
          <span>
            hPa
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
